### PR TITLE
Basic Tor support

### PR DESCRIPTION
**Pull request by nikriek**
I decided to import requesocks as a secondary networking library as it's an old fork of requests. Please **do not merge** it yet, I still have to test it out. What are you saying so far?

### CODE REVIEW

File: jodelrest.py [hunk 1]
```diff
@@ -131,7 +131,16 @@ def new_acc(self):
         self.set_pos(self.location['longtitude'], self.location['latitude'], self.location['city'])
         return "Bearer %s" % access['access_token']
 
-    def __init__(self, location, auth=None):
+    def __init__(self, location, auth=None, enable_tor=False):
+        self.tor_enabled = enable_tor
+        if self.tor_enabled:
+            import requesocks
```
Comment (nikriek) [inline jodelrest.py]: JSON parsing not included.

**Comment by jafrewa** [comment]
Can i merge ?

**Comment by nikriek** [comment]
Not yet, I still need to implement a json parser for the requesocks package. I'll do that later today.

**Comment by jafrewa** [comment]
Something like  this : 

import json

with open('data.json') as data_file:    
    data = json.load(data_file)

**Comment by jafrewa** [comment]
Also i added commenting to JodelREST so you might wanna Pull & merge for your local branch

**Comment by nikriek** [comment]
No, the problem are function calls like this:

```
return self.do_post("/users/", json_payload).json()
````

The `Response` from the requesocks package does not have a `.json()`method… My idea was to replace this with `json.loads(self.do_post("/users/", json_payload).text)` but it's not that beautiful.

**Comment by jafrewa** [comment]
Have you tried using tor as a http proxy and using the builtin proxy settings ?

**Comment by nikriek** [comment]
It does not have a SOCKS support as required for Tor.

**Comment by nikriek** [comment]
Any ideas?

**Comment by jafrewa** [comment]
How about http://stackoverflow.com/questions/10967631/how-to-make-http-request-through-a-tor-socks-proxy-using-python

**Comment by nikriek** [comment]
This example uses the urllib2 package. urllib2 is not so nice to use. That's why we have requests…

**Comment by skuom** [comment]
http://stackoverflow.com/a/2339260/849787 Have you tried this? Ought to be working with alot of socket based libraries

**Comment by jafrewa** [comment]
thats for urllib2 and afaik know urllib3 does not support SOCKS Proxys yet. We would need to have a HTTP(s)-only Proxy of TOR

**Comment by skuom** [comment]
No, it doesn't matter since Urllib3 is using sockets (difference from SOCKS). 
See this gist of how the code would look with socksipy
https://gist.github.com/skuom/53a66fdd3f462ac4e837

**Comment by jafrewa** [comment]
Can you do a pr?
Am 05.12.2015 10:55 nachm. schrieb "Marcus Gustafsson" <
notifications@github.com>:

**Pull request merged**